var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -185,13 +185,13 @@ def install(self, spec, prefix):
             or spec["blas"].name == "intel-parallel-studio"
             or spec["blas"].name == "intel-oneapi-mkl"
         ):
-            blas = "mkl_rt"
+            blas = "mkl-dynamic-lp64-iomp"
```

Comment:
intel-oneapi-mkl package provides the libraries: https://github.com/spack/spack/blob/8dc2d37447fc35fbf6a8ae3e86982f885b93ba7c/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py#L139

It does not provides the sequential libraries right now, but I could add variants to specify omp threading instead.

Another issue is that there is no intel-oneap-mkl spack package for macos. There is an mkl for macos, but there were some difficulties with integrating the installer into spack and I did not pursue it.

Reply:
Yes, this is just a hack suggested by @rgommers, it looks like it didn't work anyway so I'll revert that.

P.S. I would also love to see MKL for macOS. I used to use the DMG installer for macOS x86_64 and register it as an external package, but I couldn't find a working DMG installer for macOS arm64. A native build would be very helpful for me. In the meantime, I've had to disable MKL entirely for packages like PyTorch.

Reply:
Wanted to have a quick peek at why it did not work, but the log is gone. Do you remember why? I've had issues in conda-forge with Intel not packagling its MKL pkg-config files. That could cause it.

Reply:
In terms of build output, it's identical to the `mkl_rt` build output, can't find the library even with pkgconfig.

Reply:
I guess the reason this used to work for older numpy/scipy was that Spack explicitly passed `library_dirs`, `include_dirs`, and `libraries` via `site.cfg`. If there was support for an equivalent flag in the new meson build system, that would also work for us.

Reply:
> If the second line here doesn't work for the MKL flavor packaged in Spack, then I'd say that's a packaging bug for the Spack MKL package:

setup_dependent_build_environment is not setting PKG_CONFIG_PATH. I will fix it.

Reply:
I thought it would be enough to do:
```
    def setup_dependent_build_environment(self, env, dependent_spec):
        env.set("MKLROOT", self.component_prefix)
        env.append_path("PKG_CONFIG_PATH", self.component_prefix.lib.pkgconfig)
```

But the pc file has:
```
Requires: openmp >= 2021.0.0
```
This assumes/enforces that you will use intel compiler omp, even if you do not want the omp functionality in MKL. Any suggestions on how to handle this?

Reply:
Maybe it is ok to say that pkg-config only works for intel compiler omp. I also need to add a depends_on to intel compiler because that is where the libs are provided. I would want to make that conditional on using mkl threads=omp and using intel compiler. I also need to implement threads=omp for intel-oneapi-mkl....

Reply:
> This assumes/enforces that you will use intel compiler omp, even if you do not want the omp functionality in MKL

Maybe I'm missing something in your question, but the `pkg-config` invocation I gave explicitly asks for `-iomp`, so a dependency on `libiomp` seems expected to me. If you don't want it, you use for example:
```
$ pkg-config --libs mkl-dynamic-lp64-seq
-L/opt/intel/oneapi/mkl/2022.2.0/lib/intel64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

MKL ships 8 `.pc` files: 2x2x2 for (lp64/ilp64, dynamic/static, iomp/seq). I'm not sure who needs the static ones, but the four dynamic options may be useful to all support at least.